### PR TITLE
Schema parameters will not be compared in compatability check if conf…

### PR DESCRIPTION
**Pull request by deysman**

### CODE REVIEW

File: connect/api/src/main/java/org/apache/kafka/connect/data/SchemaProjector.java [hunk 1]
```diff
@@ -129,7 +129,7 @@ private static void checkMaybeCompatible(Schema source, Schema target) {
             throw new SchemaProjectorException("Schema type mismatch. source type: " + source.type() + " and target type: " + target.type());
         } else if (!Objects.equals(source.name(), target.name())) {
             throw new SchemaProjectorException("Schema name mismatch. source name: " + source.name() + " and target name: " + target.name());
-        } else if (!Objects.equals(source.parameters(), target.parameters())) {
+        } else if (!Objects.equals(source.parameters(), target.parameters()) && !source.parameters().containsKey("io.confluent.connect.avro.Enum")) {
```
Comment (jfb-flipp) [inline connect/api/src/main/java/org/apache/kafka/connect/data/SchemaProjector.java:132]: HAHAHAHAHA oh god

👍

**Review by jfb-flipp — APPROVED** [review]
LGTM!

File: connect/api/src/test/java/org/apache/kafka/connect/data/SchemaProjectorTest.java [hunk 2]
```diff
@@ -468,6 +468,21 @@ public void testMaybeCompatible() throws Exception {
...
+
+        Map<String, String> targetEnumParamters = new HashMap<String, String>();
+        targetEnumParamters.put("io.confluent.connect.avro.Enum", "FooEnum");
+        targetEnumParamters.put("io.confluent.connect.avro.Enum.FooSymbol", "FooSymbol");
+        targetEnumParamters.put("io.confluent.connect.avro.Enum.BarSymbol", "BarSymbol");
+        Schema targetWithEnumParameters = SchemaBuilder.string().parameters(targetEnumParamters);
+
+        Object projected = SchemaProjector.project(sourceWithEnumParameters, "FooSymbol", targetWithEnumParameters);
+
+        assertEquals("FooSymbol", projected);
```
Comment (johnnycaol) [inline connect/api/src/test/java/org/apache/kafka/connect/data/SchemaProjectorTest.java:485]: Would be nice to cover the following scenarios:
- Append enum symbol at the end (covered already)
- Add enum symbol in the middle (what happened on production)
- Remove enum symbol 
- Update enum order
- Update enum doc (what happened on production)

Comment (jfb-flipp) [reply connect/api/src/test/java/org/apache/kafka/connect/data/SchemaProjectorTest.java:485]: 👍

**Review by johnnycaol — APPROVED** [review]

File: connect/api/src/test/java/org/apache/kafka/connect/data/SchemaProjectorTest.java (hunk 2 above)
Comment (deysman) [reply connect/api/src/test/java/org/apache/kafka/connect/data/SchemaProjectorTest.java:485]: @johnnycaol At this point in the kafka connect workflow we are only dealing with string types, so different permutations of the parameters will make no difference. Those scenarios would be good to have in an integration test but I don't think that creating new integration tests is in scope.

**Pull request merged**